OpenRA.Mods.Cnc/Projectiles/IonCannon.cs

```diff
@@ -45,9 +45,18 @@ public IonCannon(Player firedBy, WeaponInfo weapon, World world, WPos launchPos,
 		public void Tick(World world)
 		{
 			anim.Tick();
+
 			if (!impacted && weaponDelay-- <= 0)
 			{
-				weapon.Impact(target, firedBy.PlayerActor, Enumerable.Empty<int>());
+				var warheadArgs = new WarheadArgs
+				{
+					Weapon = weapon,
+					Source = target.CenterPosition,
+					SourceActor = firedBy.PlayerActor,
```

Comment:
This should probably point at the advanced comm center / upgrade center that triggers the projectile.

Reply:
I faintly remember this being intentional though, probably to prevent actors from trying to retaliate to an 'attacker' in the middle of the enemy base.

Reply:
I hadn't considered that - makes sense.